#! /bin/sh /usr/share/dpatch/dpatch-run
## 25allow_symlink_sockdir.dpatch by  <hesso@pool.math.tu-berlin.de>
##
## DP: I don't know why screen should not allow a symlinked SockDir;
## DP: so now it does. (Note: this is one of TWO calls to lstat()
## DP: the whole program has - and this one isn't even wrapped in
## DP: #ifdef HAVE_LSTAT as it should.)

@DPATCH@
diff -Naur screen-4.0.3.orig/screen.c screen-4.0.3/screen.c
--- screen-4.0.3.orig/screen.c	2008-06-01 23:44:59.000000000 +0200
+++ screen-4.0.3/screen.c	2008-06-01 23:44:59.000000000 +0200
@@ -989,7 +989,7 @@
       else
 	{
 	  SockDir = SOCKDIR;
-	  if (lstat(SockDir, &st))
+	  if (stat(SockDir, &st))
 	    {
 	      n = (eff_uid == 0 && (real_uid || eff_gid == real_gid)) ? 0755 :
 	          (eff_gid != real_gid) ? 0775 :
